WeakAurasOptions/WeakAurasOptions.lua

```diff
@@ -4460,7 +4400,24 @@ function WeakAuras.UpdateDisplayButton(data)
   local button = displayButtons[id];
   if (button) then
     button:SetIcon(WeakAuras.SetThumbnail(data));
+    if WeakAurasCompanion and button:IsGroup() then
+      WeakAuras.RefreshGroupUpdateIcon(button)
+    end
+  end
+end
+
+function WeakAuras.RefreshGroupUpdateIcon(button)
+  local parentId = button.data.id
+  for id, child in pairs(displayButtons) do
```

Comment:
Instead of iterating over all displayButtons, iterate over the children of that group and check their status. That's going to be drastically less checks.